ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -194,9 +200,15 @@ let check_op_for_feature ~__context ~vmr:_ ~vmmr ~vmgmr ~power_state ~op ~ref
         some_err Api_errors.vm_lacks_feature
     | `changing_VCPUs_live when lack_feature "feature-vcpu-hotplug" ->
         some_err Api_errors.vm_lacks_feature
-    | (`suspend | `checkpoint | `pool_migrate | `migrate_send)
-      when strict && lack_feature "feature-suspend" ->
-        some_err Api_errors.vm_lacks_feature
+    | `suspend | `checkpoint | `pool_migrate | `migrate_send -> (
+      match get_feature ~vmgmr ~feature:"data-cant-suspend-reason" with
+      | Some reason ->
+          some_err Api_errors.vm_lacks_feature ~reason
```

Comment:
I think it makes sense to add a new error. I would be confused if a VM that usually suspends fine most of the time, and sometimes it fails complaining it doesn't have the feature.

Reply:
Added `vm_non_suspendable`